localization/ekf_localizer/src/ekf_localizer.cpp

```diff
@@ -151,21 +152,47 @@ void EKFLocalizer::timerCallback()
   DEBUG_INFO(get_logger(), "------------------------- end prediction -------------------------\n");
 
   /* pose measurement update */
-  if (current_pose_ptr_ != nullptr) {
+  if (!current_pose_info_queue_.empty()) {
     DEBUG_INFO(get_logger(), "------------------------- start Pose -------------------------");
+    DEBUG_INFO(
+      get_logger(), "[EKF] measurementUpdatePose number = %d",
+      int(current_pose_info_queue_.size()));
     stop_watch_.tic();
-    measurementUpdatePose(*current_pose_ptr_);
-    DEBUG_INFO(get_logger(), "[EKF] measurementUpdatePose calc time = %f [ms]", stop_watch_.toc());
-    DEBUG_INFO(get_logger(), "------------------------- end Pose -------------------------\n");
+
+    int pose_info_queue_size = int(current_pose_info_queue_.size());
+    for (int i = 0; i < pose_info_queue_size; ++i) {
+      PoseInfo pose_info = current_pose_info_queue_.front();
+      current_pose_info_queue_.pop();
+      measurementUpdatePose(*pose_info.pose, pose_info.covariance);
+      ++pose_info.counter;
+      if (pose_info.counter < pose_smoothing_steps_) {
+        current_pose_info_queue_.push(pose_info);
+      }
+    }
+    DEBUG_INFO(get_logger(), "[EKF] measurementUpdatePose calc time = %f [ms]", stop_watch_.toc()); DEBUG_INFO(get_logger(), "------------------------- end Pose
```

Comment:
Format seems to be broken

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/1027/commits/cfcbf40ca20fe1daaf8156055b9470f4ec716c99
Done :+1: